# 02 - Randomised Experiments


## The Gold Standard

In the previous session, we saw why and how association is different from causation. We also saw what is required to make association be causation.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$


To recap, association becomes causation if there is no bias. There will be no bias if $E[Y_0|T=0]=E[Y_0|T=1]$. In words, association will be causation if the treated and control are equal or comparable, except for their treatment. Or, in more technical words, when the outcome of the untreated is equal to the counterfactual outcome of the treated. Remember that this counterfactual outcome is the outcome of the treated group if they had not received the treatment.

I think we did an OK job explaining how to make association equal to causation in math terms. But that was only in theory. Now, we look at the first tool we have to make the bias vanish: **Randomised Experiments**. Randomised experiments randomly assign individuals in a population to a treatment or to a control group. The proportion that receives the treatment doesn't have to be 50%. You could have an experiment where only 10% of your samples get the treatment.

Randomisation annihilates bias by making the potential outcomes independent of the treatment.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

This can be confusing at first (it was for me). But don't worry, my brave and true fellow, I'll explain it further. If the outcome is independent of the treatment, doesn't this also imply that the treatment has no effect? Well, yes! But notice I'm not talking about the outcomes. Instead, I'm talking about the **potential** outcomes. The potential outcome is how the outcome **would have been** under treatment ($Y_1$) or under control ($Y_0$). In randomized trials, we **don't** want the outcome to be **independent** of the treatment since we think **the treatment causes** the outcome. But we want the **potential** outcomes to be independent of the treatment. 

![img](./data/img/rct/indep.png)

Saying that the potential outcomes are independent of the treatment is saying that they would be, in expectation, the same in the treatment or the control group. In simpler terms, it means that treatment and control groups are comparable. Or that knowing the treatment assignment doesn't give me any information on how the outcome was previous to the treatment. Consequently, $(Y_0, Y_1)\perp T$ means that the treatment is the only thing generating a difference between the outcome in the treated and in the control group. To see this, notice that independence implies precisely that

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

Which, as we've seen, makes it so that

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

So, randomization gives us a way to use a simple difference in means between treatment and control and call that the treatment effect.


## In a School Far, Far Away

In 2020, the Coronavirus Pandemic forced businesses to adapt to social distancing. Delivery services became widespread, and big corporations shifted to a remote work strategy. With schools, it wasn't different. Many started their own online repository of classes. 

Four months into the crisis, many wonder if the introduced changes could be maintained. There is no question that online learning has its benefits. It is cheaper since it can save on real estate and transportation. It can also be more digital, leveraging world-class content from around the globe, not just from a fixed set of teachers. Despite all of that, we still need to answer if online learning has a negative or positive impact on the student's academic performance.

One way to answer this is to take students from schools that give mostly online classes and compare them with students from schools that provide lectures in traditional classrooms. As we know by now, this is not the best approach. It could be that online schools attract only the well-disciplined students that do better than average even if the class were in-person. In this case, we would have a positive bias, where the treated are academically better than the untreated: $E[Y_0|T=1] > E[Y_0|T=0]$.

Or on the flip side, it could be that online classes are cheaper and are composed chiefly of less wealthy students, who might have to work besides studying. In this case, these students would do worse than those from the in-person schools even if they took in-person classes. If this was the case, we would have a bias in the other direction, where the treated are academically worse than the untreated:  $E[Y_0|T=1] < E[Y_0|T=0]$. 

So, although we could make simple comparisons, it wouldn't be compelling. One way or another, we could never be sure if there wasn't any bias lurking around and masking our causal effect.

![img](./data/img/rct/lurking_bias.png)

To solve that, we need to make the treated and untreated comparable $E[Y_0|T=1] = E[Y_0|T=0]$. One way to force this is by randomly assigning the online and in-person classes to students. If we managed to do that, the treatment and untreated would be, on average, the same, except for the treatment they receive. 

Fortunately, some economists have done that for us. They've randomized classes so that some students were assigned to have face-to-face lectures, others to have only online lessons, and a third group to have a blended format of both online and face-to-face classes. They collected data on a standard exam at the end of the semester.

Here is what the data looks like:

# 02 - Esperimenti Randomizzati

## Lo Standard Aureo

Nella sessione precedente, abbiamo visto perché e come l'associazione è diversa dalla causalità. Abbiamo anche visto cosa è richiesto per rendere l'associazione uguale alla causalità.

$
E[Y|T=1] - E[Y|T=0] = \underbrace{E[Y_1 - Y_0|T=1]}_{ATT} + \underbrace{\{ E[Y_0|T=1] - E[Y_0|T=0] \}}_{BIAS}
$

Per ricapitolare, l' **associazione diventa causalità se non c'è distorsione**. Non ci sarà distorsione se $E[Y_0|T=0]=E[Y_0|T=1]$. In parole semplici, l'associazione diventerà causalità **se i trattati e il controllo sono uguali o comparabili**, eccetto per il trattamento che ricevono. Oppure, in termini più tecnici, quando l'esito dei non trattati è uguale all'esito controfattuale dei trattati. Ricorda che questo esito controfattuale è l'esito del gruppo trattato se non avessero ricevuto il trattamento.

Penso che abbiamo fatto un buon lavoro nel spiegare come rendere l'associazione uguale alla causalità in termini matematici. Ma era solo in teoria. Ora, guardiamo il primo strumento che abbiamo per far sparire la distorsione: **Gli Esperimenti Randomizzati**. Gli esperimenti randomizzati **assegnano casualmente gli individui di una popolazione** a un trattamento o a un gruppo di controllo. La proporzione che riceve il trattamento **non deve necessariamente essere del 50%**. Potresti avere un esperimento in cui solo il 10% dei tuoi campioni riceve il trattamento.

La randomizzazione annichilisce la distorsione rendendo i risultati potenziali indipendenti dal trattamento.

$
(Y_0, Y_1) \perp\!\!\!\perp T
$

[n.d.t. il simbolo $\perp\!\!\!\perp$  indica l'indipendenza statistica: Dire che A e B sono indipendenti (A⊥⊥B) significa che conoscere il valore di una non ti dà assolutamente alcuna informazione sul valore dell'altra.]

Questo può essere confuso all'inizio (lo è stato per me). Ma non preoccuparti, mio coraggioso e sincero collega, lo spiegherò ulteriormente. Se l'esito è indipendente dal trattamento, ciò non implica anche che il trattamento non ha effetto? Beh, sì! Ma nota che non sto parlando degli esiti. Invece, sto parlando dei risultati **potenziali**. Il risultato potenziale è come l'esito **sarebbe stato** sotto trattamento ($Y_1$) o sotto controllo ($Y_0$). Nei trial randomizzati, **non** vogliamo che l'esito sia **indipendente** dal trattamento poiché pensiamo che **il trattamento causa** l'esito. Ma vogliamo che i risultati **potenziali** siano indipendenti dal trattamento.

![img](./data/img/rct/indep.png)

Dire che i risultati potenziali sono indipendenti dal trattamento significa dire che sarebbero, in aspettativa, gli stessi nel gruppo trattato o nel gruppo di controllo. In termini più semplici, significa che i gruppi trattati e di controllo sono comparabili. Oppure che conoscere l'assegnazione del trattamento non mi dà alcuna informazione su come l'esito era precedente al trattamento. Di conseguenza, $(Y_0, Y_1)\perp T$ significa che il trattamento è l'unica cosa che genera una differenza tra l'esito nei trattati e nel gruppo di controllo. Per vedere questo, nota che l'indipendenza implica precisamente che

$
E[Y_0|T=0]=E[Y_0|T=1]=E[Y_0]
$

Che, come abbiamo visto, fa sì che

$
E[Y|T=1] - E[Y|T=0] = E[Y_1 - Y_0]=ATE
$

Quindi, la randomizzazione ci dà un modo per usare una semplice differenza nelle medie tra il trattamento e il controllo e chiamare quello l'effetto del trattamento.

## In una Scuola Molto, Molto Lontana

Nel 2020, la pandemia di Coronavirus ha costretto le aziende ad adattarsi al distanziamento sociale. I servizi di consegna sono diventati diffusi e le grandi aziende si sono spostate verso una strategia di lavoro remoto. Con le scuole, non è stato diverso. Molte hanno avviato i loro stessi repository online di lezioni.

Quattro mesi dopo l'inizio della crisi, molti si chiedono se i cambiamenti introdotti potrebbero essere mantenuti. Non c'è dubbio che l'apprendimento online ha i suoi vantaggi. È più economico poiché può risparmiare su immobili e trasporti. Può anche essere più digitale, sfruttando contenuti di classe mondiale da tutto il mondo, non solo da un insieme fisso di insegnanti. Nonostante tutto ciò, abbiamo ancora bisogno di rispondere se l'apprendimento online ha un impatto negativo o positivo sulle prestazioni accademiche dello studente.

Un modo per rispondere è prendere studenti da scuole che offrono principalmente lezioni online e confrontarli con studenti da scuole che offrono lezioni in aule tradizionali. Come sappiamo ormai, questo non è il miglior approccio. Potrebbe accadere che le scuole online attraggono solo studenti ben disciplinati che ottengono risultati migliori della media anche se la classe fosse in presenza. In questo caso, avremmo una distorsione positiva, dove i trattati sono accademicamente migliori dei non trattati: $E[Y_0|T=1] > E[Y_0|T=0]$.

O al contrario, potrebbe accadere che le lezioni online siano più economiche e siano composte principalmente da studenti meno ricchi, che potrebbero dover lavorare oltre a studiare. In questo caso, questi studenti farebbero peggio rispetto a quelli delle scuole in presenza anche se seguissero lezioni in presenza. Se fosse questo il caso, avremmo una distorsione nella direzione opposta, dove i trattati sono accademicamente peggiori dei non trattati: $E[Y_0|T=1] < E[Y_0|T=0]$.

Quindi, sebbene potessimo fare semplici confronti, non sarebbe convincente. In un modo o nell'altro, non potremmo mai essere sicuri che non ci fosse una distorsione nascosta che mascherasse il nostro effetto causale.

![img](./data/img/rct/lurking_bias.png)

Per risolvere questo, abbiamo bisogno di rendere i trattati e i non trattati comparabili $E[Y_0|T=1] = E[Y_0|T=0]$. Un modo per forzare questo è assegnare casualmente le lezioni online e in presenza agli studenti. Se riuscissimo a farlo, il trattamento e i non trattati sarebbero, in media, uguali, ad eccezione del trattamento che ricevono.

Fortunatamente, alcuni economisti l'hanno fatto per noi. Hanno randomizzato le lezioni in modo che alcuni studenti fossero assegnati ad avere lezioni faccia a faccia, altri avessero solo lezioni online, e un terzo gruppo avesse un formato ibrido di lezioni sia online che faccia a faccia. Hanno raccolto dati su un esame standardizzato alla fine del semestre.

Ecco come si presenta il dataset:

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv("./data/online_classroom.csv")
print(data.shape)
data.head()

(323, 10)


,gender,asian,black,hawaiian,hispanic,unknown,white,format_ol,format_blended,falsexam
0,0,0.0,0.0,0.0,0.0,0.0,1.0,0,0.0,63.29997
1,1,0.0,0.0,0.0,0.0,0.0,1.0,0,0.0,79.96000
2,1,0.0,0.0,0.0,0.0,0.0,1.0,0,1.0,83.37000
3,1,0.0,0.0,0.0,0.0,0.0,1.0,0,1.0,90.01994
4,1,0.0,0.0,0.0,0.0,0.0,1.0,1,0.0,83.30000


We can see that we have 323 samples. It's not exactly big data, but something we can work with. To estimate the causal effect, we can simply compute the mean score for each of the treatment groups.

Possiamo vedere che abbiamo 323 campioni. Non è esattamente big data, ma qualcosa con cui possiamo lavorare. Per stimare l'effetto causale, possiamo semplicemente calcolare il punteggio medio per ogni gruppo di trattamento.

In [4]:
(data
 .assign(class_format = np.select(
     [data["format_ol"].astype(bool), data["format_blended"].astype(bool)],
     ["online", "blended"],
     default="face_to_face"
 ))
 .groupby(["class_format"])
 .mean())

,gender,asian,black,hawaiian,hispanic,unknown,white,format_ol,format_blended,falsexam
class_format,,,,,,,,,,
blended,0.550459,0.217949,0.102564,0.025641,0.012821,0.012821,0.628205,0.0,1.0,77.093731
face_to_face,0.633333,0.202020,0.070707,0.000000,0.010101,0.000000,0.717172,0.0,0.0,78.547485
online,0.542553,0.228571,0.028571,0.014286,0.028571,0.000000,0.700000,1.0,0.0,73.635263


Yup. It's that simple. We can see that face-to-face classes yield a 78.54 average score, while online courses yield a 73.63 average score. Not so good news for the proponents of online learning. The $ATE$ for an online class is thus -4.91. This means that **online classes cause students to perform about 5 points lower**, on average. That's it. You don't need to worry that online courses might have poorer students that can't afford face-to-face classes or, for that matter, you don't have to worry that the students from the different treatments are different in any way other than the treatment they received. By design, the random experiment is made to wipe out those differences. 

For this reason, a good sanity check to see if the randomisation was done right (or if you are looking at the correct data) is to check if the treated are equal to the untreated in pre-treatment variables. Our data has information on gender and ethnicity to see if they are similar across groups. We can say that they look pretty similar for the `gender`, `asian`, `hispanic`, and `white` variables. The `black` variable, however, seems a little bit different. This draws attention to what happens with a small dataset. Even under randomisation, it could be that, by chance, one group is different from another. In large samples, this difference tends to disappear.

## The Ideal Experiment

Randomised experiments or Randomised Controlled Trials (RCT) are the most reliable way to get causal effects. It's a straightforward technique and absurdly convincing. It is so powerful that most countries have it as a requirement for showing the effectiveness of new medicine. To make a terrible analogy, you can think of RCT as Aang, from Avatar: The Last Airbender, while other techniques are more like Sokka. Sokka is cool and can pull some neat tricks here and there, but Aang can bend the four elements and connect with the spiritual world. Think of it this way, if we could, RCT would be all we would ever do to uncover causality. A well designed RCT is the dream of any scientist.

![img](./data/img/rct/science_dream.png)

Unfortunately, they tend to be either very expensive or just plain unethical. Sometimes, we simply can't control the assignment mechanism. Imagine yourself as a physician trying to estimate the effect of smoking during pregnancy on baby weight at birth. You can't simply force a random portion of moms to smoke during pregnancy. Or say you work for a big bank, and you need to estimate the impact of the credit line on customer churn. It would be too expensive to give random credit lines to your customers. Or that you want to understand the impact of increasing the minimum wage on unemployment. You can't simply assign countries to have one or another minimum wage. You get the point.

We will later see how to lower the randomisation cost by using conditional randomisation, but there is nothing we can do about unethical or unfeasible experiments. Still, whenever we deal with causal questions, it is worth thinking about the **ideal experiment**. Always ask yourself, if you could, **what would be the perfect experiment you would run to uncover this causal effect?** This tends to shed some light on the way how we can discover the causal effect even without the ideal experiment.


## The Assignment Mechanism

In a randomised experiment, the mechanism that assigns units to one treatment or the other is, well, random. As we will see later, all causal inference techniques will somehow try to identify the assignment mechanisms of the treatments. When we know for sure how this mechanism behaves, causal inference will be much more confident, even if the assignment mechanism isn't random.

Unfortunately, the assignment mechanism can't be discovered by simply looking at the data. For example, if you have a dataset where higher education correlates with wealth, you can't know for sure which one caused which by just looking at the data. You will have to use your knowledge about how the world works to argue in favor of a plausible assignment mechanism: is it the case that schools educate people, making them more productive and leading them to higher-paying jobs. Or, if you are pessimistic about education, you can say that schools do nothing to increase productivity, and this is just a spurious correlation because only wealthy families can afford to have a kid get a higher degree.

In causal questions, we usually can argue in both ways: that X causes Y, or that it is a third variable Z that causes both X and Y, and hence the X and Y correlation is just spurious. For this reason, knowing the assignment mechanism leads to a much more convincing causal answer. This is also what makes causal inference so exciting. While applied ML is usually just pressing some buttons in the proper order, applied causal inference requires you to seriously think about the mechanism generating that data. 


## Key Ideas

We looked at how randomised experiments are the simplest and most effective way to uncover causal impact. It does this by making the treatment and control groups comparable. Unfortunately, we can't do randomised experiments all the time, but it is still helpful to think about what is the ideal experiment we would do if we could.

Someone familiar with statistics might be protesting right now that I didn't look at the variance of my causal effect estimate. How can I know that a 4.91 points decrease is not due to chance? In other words, how can I know if the difference is statistically significant? And they would be right. Don't worry. I intend to review some statistical concepts next. 


## References

I like to think of this entire book as a tribute to Joshua Angrist, Alberto Abadie and Christopher Walters for their amazing Econometrics class. Most of the ideas here are taken from their classes at the American Economic Association. Watching them is what is keeping me sane during this tough year of 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

I'll also like to reference the amazing books from Angrist. They have shown me that Econometrics, or 'Metrics as they call it, is not only extremely useful but also profoundly fun.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

My final reference is Miguel Hernan and Jamie Robins' book. It has been my trustworthy companion in the most thorny causal questions I had to answer.

* [Causal Inference Book](https://miguelhernan.org/whatifbook)

The data used here is from a study of Alpert, William T., Kenneth A. Couch, and Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribute

Causal Inference for the Brave and True is an open-source material on causal inference, the statistics of science. Its goal is to be accessible monetarily and intellectually. It uses only free software based on Python.
If you found this book valuable and want to support it, please go to [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). If you are not ready to contribute financially, you can also help by fixing typos, suggesting edits, or giving feedback on passages you didn't understand. Go to the book's repository and [open an issue](https://github.com/matheusfacure/python-causality-handbook/issues). Finally, if you liked this content, please share it with others who might find it helpful and give it a [star on GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).

Esatto. È così semplice. Possiamo vedere che le lezioni faccia a faccia producono un punteggio medio di 78,54, mentre i corsi online producono un punteggio medio di 73,63. Non è una buona notizia per i sostenitori dell'apprendimento online. L'$ATE$ per una lezione online è quindi -4,91. Ciò significa che **le lezioni online causano agli studenti di ottenere prestazioni circa 5 punti inferiori**, in media. Punto. Non devi preoccuparti che i corsi online potrebbero avere studenti più poveri che non possono permettersi lezioni faccia a faccia o, per questo, non devi preoccuparti che gli studenti dei diversi trattamenti siano diversi in alcun modo diverso dal trattamento che hanno ricevuto. Per progettazione, l'esperimento casuale è fatto per cancellare quelle differenze.

Per questo motivo, un buon controllo di sanità mentale per vedere se la randomizzazione è stata fatta correttamente (o se stai guardando i dati corretti) è verificare se i trattati sono uguali ai non trattati nelle variabili pre-trattamento. I nostri dati hanno informazioni su genere ed etnia per vedere se sono simili tra i gruppi. Possiamo dire che sembrano abbastanza simili per le variabili `gender`, `asian`, `hispanic` e `white`. La variabile `black`, tuttavia, sembra un po' diversa. Questo attira l'attenzione su cosa succede con un piccolo dataset. Anche sotto randomizzazione, potrebbe accadere che, per caso, un gruppo sia diverso da un altro. In campioni grandi, questa differenza tende a scomparire.

## L'Esperimento Ideale

Gli esperimenti randomizzati o Trial Controllati Randomizzati (RCT) sono il modo più affidabile per ottenere effetti causali. È una tecnica diretta e incredibilmente convincente. È così potente che la maggior parte dei paesi lo richiede per dimostrare l'efficacia di nuovi medicinali. Per fare un'analogia terribile, puoi pensare all'RCT come Aang, da Avatar: L'Ultimo Airbender, mentre altre tecniche sono più come Sokka. Sokka è cool e può fare alcuni trucchi ordinati qua e là, ma Aang può piegare i quattro elementi e connettersi con il mondo spirituale. Pensala così, se potessimo, l'RCT sarebbe tutto quello che faremmo per scoprire la causalità. Un RCT ben progettato è il sogno di qualsiasi scienziato.

![img](./data/img/rct/science_dream.png)

Sfortunatamente, tendono ad essere molto costosi o semplicemente non etici. A volte, semplicemente non possiamo controllare il meccanismo di assegnazione. Immagina di essere un medico che cerca di stimare l'effetto del fumo durante la gravidanza sul peso del bambino alla nascita. Non puoi semplicemente costringere una porzione casuale di mamme a fumare durante la gravidanza. Oppure supponiamo che lavori per una grande banca e devi stimare l'impatto della linea di credito sulla perdita di clienti. Sarebbe troppo costoso dare linee di credito casuali ai tuoi clienti. O che tu voglia capire l'impatto dell'aumento del salario minimo sulla disoccupazione. Non puoi semplicemente assegnare paesi ad avere un salario minimo o un altro. Afferri il concetto.

Vedremo più tardi come abbassare il costo della randomizzazione utilizzando la randomizzazione condizionata, ma non c'è niente che possiamo fare riguardo agli esperimenti non etici o fattibili. Tuttavia, ogni volta che affrontiamo domande causali, vale la pena pensare all'**esperimento ideale**. Chiediti sempre, se potessi, **quale sarebbe l'esperimento perfetto che realizzeresti per scoprire questo effetto causale?** Questo tende a fare luce su come possiamo scoprire l'effetto causale anche senza l'esperimento ideale.

## Il Meccanismo di Assegnazione

In un esperimento randomizzato, il meccanismo che assegna le unità a un trattamento o a un altro è, beh, casuale. Come vedremo più tardi, tutte le tecniche di inferenza causale cercheranno in qualche modo di identificare i meccanismi di assegnazione dei trattamenti. Quando sappiamo con certezza come si comporta questo meccanismo, l'inferenza causale sarà molto più sicura, anche se il meccanismo di assegnazione non è casuale.

Sfortunatamente, il meccanismo di assegnazione non può essere scoperto semplicemente guardando i dati. Ad esempio, se hai un dataset in cui l'istruzione superiore è correlata con la ricchezza, non puoi sapere con certezza quale ha causato quale guardando solo i dati. Dovrai usare la tua conoscenza su come funziona il mondo per argomentare a favore di un meccanismo di assegnazione plausibile: è il caso che le scuole educano le persone, rendendole più produttive e portandole a lavori meglio pagati. Oppure, se sei pessimista riguardo all'istruzione, puoi dire che le scuole non fanno nulla per aumentare la produttività, ed è solo una correlazione spuria perché solo le famiglie ricche possono permettersi che un bambino ottenga una laurea.

Nelle domande causali, di solito possiamo argomentare in entrambi i modi: che X causa Y, oppure che sia una terza variabile Z che causa sia X che Y, e quindi la correlazione tra X e Y è solo spuria. Per questo motivo, conoscere il meccanismo di assegnazione porta a una risposta causale molto più convincente. Questo è anche ciò che rende l'inferenza causale così entusiasmante. Mentre il ML applicato di solito è solo premere alcuni pulsanti nell'ordine appropriato, l'inferenza causale applicata ti richiede di pensare seriamente al meccanismo che genera quei dati.

## Idee Chiave

Abbiamo visto come gli esperimenti randomizzati siano il modo più semplice e più efficace per scoprire l'impatto causale. Lo fa rendendo i gruppi di trattamento e controllo comparabili. Sfortunatamente, non possiamo sempre fare esperimenti randomizzati, ma è comunque utile pensare a quale sarebbe l'esperimento ideale che faremmo se potessimo.

Qualcuno che ha familiarità con la statistica potrebbe stare protestando proprio ora perché non ho esaminato la varianza della mia stima dell'effetto causale. Come posso sapere che una diminuzione di 4,91 punti non è dovuta al caso? In altre parole, come posso sapere se la differenza è statisticamente significativa? E avrebbero ragione. Non preoccuparti. Ho intenzione di ripassare alcuni concetti statistici dopo.

## Riferimenti

Mi piace pensare a questo intero libro come un tributo a Joshua Angrist, Alberto Abadie e Christopher Walters per il loro fantastico corso di Econometria. La maggior parte delle idee qui sono tratte dai loro corsi presso l'American Economic Association. Guardarli è ciò che mi mantiene sano di mente durante questo difficile anno 2020.
* [Cross-Section Econometrics](https://www.aeaweb.org/conference/cont-ed/2017-webcasts)
* [Mastering Mostly Harmless Econometrics](https://www.aeaweb.org/conference/cont-ed/2020-webcasts)

Vorrei anche fare riferimento ai fantastici libri di Angrist. Mi hanno dimostrato che l'Econometria, o 'Metrics come la chiamano loro, non è solo estremamente utile ma anche profondamente divertente.

* [Mostly Harmless Econometrics](https://www.mostlyharmlesseconometrics.com/)
* [Mastering 'Metrics](https://www.masteringmetrics.com/)

Il mio ultimo riferimento è il libro di Miguel Hernan e Jamie Robins. È stato il mio fedele compagno nelle domande causali più spinose a cui ho dovuto rispondere.

* [Causal Inference Book](https://miguelhernan.org/whatifbook)

I dati utilizzati qui provengono da uno studio di Alpert, William T., Kenneth A. Couch, e Oskar R. Harmon. 2016. ["A Randomized Assessment of Online Learning"](https://www.aeaweb.org/articles?id=10.1257/aer.p20161057). American Economic Review, 106 (5): 378-82.

![img](./data/img/poetry.png)

## Contribuisci

Causal Inference for the Brave and True è un materiale open-source sull'inferenza causale, la statistica della scienza. Il suo obiettivo è essere accessibile sia economicamente che intellettualmente. Utilizza solo software libero basato su Python.
Se hai trovato questo libro prezioso e vuoi supportarlo, vai su [Patreon](https://www.patreon.com/causal_inference_for_the_brave_and_true). Se non sei pronto a contribuire finanziariamente, puoi anche aiutare correggendo errori di battitura, suggerendo modifiche o dando feedback su passaggi che non hai capito. Vai al repository del libro e [apri una issue](https://github.com/matheusfacure/python-causality-handbook/issues). Infine, se ti è piaciuto questo contenuto, condividilo con altri che potrebbero trovarlo utile e dagli una [stella su GitHub](https://github.com/matheusfacure/python-causality-handbook/stargazers).